src/components/Admin/Trainer/ControllerDataset.js

```diff
@@ -25,51 +22,82 @@ export class ControllerDataset {
     return itemObj;
   }
 
-  async addExamples(examples) {
-    if (examples.length < 1) {
-      this.setReadyStatus('Please provide at least one image.');
-      return;
-    }
+  getCollectionReference = async name => {
+    return await this.db.collection(name);
+  };
+
+  getItemReference = async label => {
+    return await this.db.collection('item_data').doc(label);
+  };
+
+  getItemCount = async itemReference => {
+    return await itemReference
+      .get()
```

Comment:
Just a general comment that doesn't need addressing, we could return the promises directly if we didn't want to use async/await

```
getItemCount = itemReference => itemReference.get().then(snapshot => snapshot.data().count).catch(() => 0)